json/test.zig

```diff
@@ -0,0 +1,76 @@
+const std = @import("std");
+
+const Coordinate = struct {
+    x: f64,
+    y: f64,
+    z: f64,
+
+    fn eql(left: Coordinate, right: Coordinate) bool {
+        return left.x == right.x and left.y == right.y and left.z == right.z;
+    }
+};
+
+const TestStruct = struct {
+    coordinates: []Coordinate,
+};
+
+fn notify(msg: []const u8) void {
+    const addr = std.net.Address.parseIp("127.0.0.1", 9001) catch unreachable;
+    var stream = std.net.tcpConnectToAddress(addr) catch unreachable;
+    _ = stream.write(msg) catch unreachable;
+    stream.close();
+}
+
+fn readFile(alloc: *std.mem.Allocator, filename: []const u8) ![]const u8 {
+    const file = try std.fs.cwd().openFile(filename, std.fs.File.OpenFlags{});
+    defer file.close();
+
+    const size = try file.getEndPos();
+    const text = try alloc.alloc(u8, size);
+    _ = try file.readAll(text);
+    return text;
+}
+
+fn calc(alloc: *std.mem.Allocator, text: []const u8) Coordinate {
+    var stream = std.json.TokenStream.init(text);
+    const opts = std.json.ParseOptions{ .allocator = alloc, .ignore_unknown_fields = true };
```

Comment:
`ignore_unknown_fields` is not required.

Reply:
It is or the parsing will fail. The generated JSON file contains extra fields (e.g. "name", "opts"), so not ignoring them makes the parse function return error, but since we are telling this cannot happen (unreachable), the program segfaults.